### PR TITLE
Import well-known-type file struct.proto

### PR DESCRIPTION
**Pull request by bkuehnert**
This commit adds the protos defined in google/protobuf/struct.proto. It
also adds the special JSON mappings that are specified for these protos.

Additionally, this commit adds the parsing utility function
expect-token-or-string, which functions like  expect-char, but it uses
the parsing function parse-token-or-string instead of next-char.

This is a copy of #247 (which was sadly abandoned due to git issues)

### CODE REVIEW

File: tests/json-test.lisp [hunk 1]
```diff
@@ -152,6 +152,32 @@ the result is PROTO-EQUAL with MSG."
...
+         (list-val (google:make-value :list-value list-val-proto))
+         (struct1 (google:make-struct)))
+    (setf (google:struct.fields-gethash "a" struct1) null-val)
+    (setf (google:struct.fields-gethash "b" struct1) num-val)
+    (setf (google:struct.fields-gethash "c" struct1) str-val)
+    (setf (google:struct.fields-gethash "d" struct1) bool-val)
+    (setf (google:struct.fields-gethash "e" struct1) list-val)
+    (let ((struct2 (google:make-struct))
+          (list-proto2 (google:make-list-value :values (list (google:make-value :struct-value struct1)))))
+      (setf (google:struct.fields-gethash "a" struct2) (google:make-value :struct-value struct1))
```
Comment (Slids) [inline tests/json-test.lisp:170]: new line

File: tests/json-test.lisp [hunk 2]
```diff
@@ -152,6 +152,32 @@ the result is PROTO-EQUAL with MSG."
...
+      (setf (google:struct.fields-gethash "b" struct2) (google:make-value :list-value list-proto2))
```
Comment (Slids) [inline tests/json-test.lisp:171]: new line

File: tests/json-test.lisp [hunk 3]
```diff
@@ -152,6 +152,32 @@ the result is PROTO-EQUAL with MSG."
...
+         (list-val-proto (google:make-list-value :values (list null-val num-val str-val bool-val)))
+         (list-val (google:make-value :list-value list-val-proto))
+         (struct1 (google:make-struct)))
+    (setf (google:struct.fields-gethash "a" struct1) null-val)
+    (setf (google:struct.fields-gethash "b" struct1) num-val)
+    (setf (google:struct.fields-gethash "c" struct1) str-val)
+    (setf (google:struct.fields-gethash "d" struct1) bool-val)
+    (setf (google:struct.fields-gethash "e" struct1) list-val)
+    (let ((struct2 (google:make-struct))
+          (list-proto2 (google:make-list-value :values (list (google:make-value :struct-value struct1)))))
```
Comment (Slids) [inline tests/json-test.lisp:169]: new line

File: tests/json-test.lisp [hunk 4]
```diff
@@ -152,6 +152,32 @@ the result is PROTO-EQUAL with MSG."
   (let* ((field-mask (google:make-field-mask :paths '("user.display_name" "photo"))))
     (json-roundtrip field-mask)))
 
+(deftest struct-roundtrip (json-suite)
+  (let* ((null-val (google:make-value :null-value :null-value))
+         (num-val (google:make-value :number-value 1.5d0))
+         (str-val (google:make-value :string-value "test"))
+         (bool-val (google:make-value :bool-value t))
+         (list-val-proto (google:make-list-value :values (list null-val num-val str-val bool-val)))
```
Comment (Slids) [inline tests/json-test.lisp:160]: new lien

File: tests/json-test.lisp (hunk 2 above)
Comment (bkuehnert) [reply tests/json-test.lisp:172]: this one is < 100 columns

**Pull request merged**